integrators/methods.hpp

```diff
@@ -0,0 +1,1007 @@
+  static constexpr CompositionMethod composition = BA;
+  static constexpr serialization::FixedStepSizeIntegrator::Kind kind =
+      serialization::FixedStepSizeIntegrator::
+          MCLACHLAN_ATELA_1992_ORDER_4_OPTIMAL;
+  static constexpr int stages = stages(evaluations, composition);
+  static constexpr FixedVector<double, stages> a{{{0.5153528374311229364,
+                                                   -0.085782019412973646,
+                                                   0.4415830236164665242,
+                                                   0.1288461583653841854}}};
+  static constexpr FixedVector<double, stages> b{{{0.1344961992774310892,
+                                                   -0.2248198030794208058,
+                                                   0.7563200005156682911,
+                                                   0.3340036032863214255}}};
+};
+struct McLachlanAtela1992Order5Optimal : SymplecticRungeKuttaNyström {
+  static constexpr int order = 5;
+  static constexpr bool time_reversible = false;
+  static constexpr int evaluations = 6;
+  static constexpr CompositionMethod composition = BA;
+  static constexpr serialization::FixedStepSizeIntegrator::Kind kind =
+      serialization::FixedStepSizeIntegrator::
+          MCLACHLAN_ATELA_1992_ORDER_5_OPTIMAL;
+  static constexpr int stages = stages(evaluations, composition);
+  static constexpr FixedVector<double, stages> a{{{0.339839625839110000,
+                                                   -0.088601336903027329,
+                                                   0.5858564768259621188,
+                                                   -0.603039356536491888,
+                                                   0.3235807965546976394,
+                                                   0.4423637942197494587}}};
+  static constexpr FixedVector<double, stages> b{{{0.1193900292875672758,
+                                                   0.6989273703824752308,
+                                                   -0.1713123582716007754,
+                                                   0.4012695022513534480,
+                                                   0.0107050818482359840,
+                                                   -0.0589796254980311632}}};
+};
+
+// This integrator goes by many names, see Hairer, Lubich, and Wanner (2003),
+// Geometric numerical integration illustrated by the Störmer–Verlet method,
+// http://www.math.kit.edu/ianm3/lehre/geonumint2009s/media/gni_by_stoermer-verlet.pdf,
+// section 1.3 (Historical remarks).
+// Notably, it appears in Philosophiae Naturalis Principia Mathematica,
+// in section II (De inventione Virium centripetarum), in the proof of
+// theorem I.  See p. 37 of the first edition,
+// https://cudl.lib.cam.ac.uk/view/PR-ADV-B-00039-00001/97.
+// It also appears in:
+// - Delambre (1790), De l'usage du calcul differentiel dans la
+//   construction des tables astronomiques.
```

Comment:
in Mémoires de l'Académie Royale des Sciences de Turin, vol. V, Mémoires présentés à l'Académie, p. 143-180.
Since we have PDFs of the whole book rather than individual articles, it is hard to find things without page numbers.

Reply:
Done.